docs/kbs_attestation_protocol.md

```diff
@@ -108,113 +146,303 @@ attestation challenge to KBC. The payload format is as follows:
+KBS uses the HTTP transport protocol to exchange the above described
+attestation messages, as HTTP requests and responses payloads.
 
-- `token`
+The first step in the HTTP-based KBS attestation protocol described here allows
+for an attester to authenticate itself against a trusted Key Brokering Service.
+The second step is when an attester requests one or more protected resources
+from the KBS. The resources are delivered by the KBS back to the attester based
+on the HTTP Cookie bound authentication results.
 
-If the attestation evidence sent by KBC is verified before returning the response, KBS will issue 
-a token to KBC in this field and sign it with KBS's root key. KBS will maintain the corresponding 
-relationship between each issued token and its original attestation result. KBC can use this token 
-to request resources or services from KBS without Challenge and Attestation within the validity period of 
-the token. If there is already a token in the KBC's request, no new token will be issued and this 
-field will be left blank.
+## Authentication
 
-The [JSON web token](https://jwt.io/) standard is adopted for the token. The format is as follows:
+Before being able to request a protected resource (a key, a token, etc), an
+attester must authenticate itself against the remote KBS. Only after succesfully
+authenticated itself, it can request those resources.
 
-(Note that the `/*...*/` comments are not valid in JSON web token, and must not be used in real token.)
+The authentication service is provided by the KBS through two endpoints:
+
+1. `/kbs/v0/auth` only accepts `POST` requests which body is a [KBS Request](#request)
+   JSON payload. This endpoint is for the attester to initiate the attestation
+   protocol and authenticate itself against the KBS. The KBS reply HTTP response
+   is composed of:
+   a. A `Set-Cookie` header set to `kbs-session-id=<session>`. The KBS tracks
+   the attester and its attestation results with that cookie.
+   b. An attestation challenge for the attester to take. This is the content of
+   the response, set to a [KBS Challenge](#challenge) JSON payload.
+2. `/kbs/v0/attest` only accepts `POST` requests which body is a [KBS Attestation](#attestation)
+   JSON payload and the header contains a `Cookie` set to the value received in
+   step 1.a. This is how the attester replies to attestation challenge received
+   in step 1.b. The KBS replies to that request with an empty HTTP response
+   (no content), which HTTP status indicates if the attestation was succesful or
+   not.
+
+```
+┌──────────┐                                          ┌─────┐
+│ Attester │                                          │ KBS │
+└────┳─────┘                                          └──┳──┘
+     ┃                 ┌─────────────────┐               ┃
+     ┃─────────────────┼POST /kbs/v0/auth┼──────────────▶┃
+     ┃                 ├─────────────────┤               ┃
+     ┃                 │Header: N/A      │               ┃
```

Comment:
When you say "n/a" do you mean literally empty, or just nothing of interest to show here?
For example, are `Content-Length` and `Content-Type` required? What about a date? If specified, is that validated (what if the date was the epoch or a value far in the future)?

Reply:
I mean that the header should not contain any kbs related cookie (as opposed  to the next ones).
I tried to fix it now, let me know if that looks better.